# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --prefix /wl

# build
make

# install
install -d $DESTDIR/wl/include
install -m 644 gbm.h $DESTDIR/wl/include
install -d $DESTDIR/wl/lib
install libgbm.so.1 $DESTDIR/wl/lib/libstubgbm.so.1
ln -sf libstubgbm.so.1 $DESTDIR/wl/lib/libgbm.so
install -d $DESTDIR/wl/lib/pkgconfig
install -m 644 gbm.pc $DESTDIR/wl/lib/pkgconfig

# build.pc
install -d $DESTDIR/wl/share/pkgconfig
cat > $DESTDIR/wl/share/pkgconfig/libgbm-stub-build.pc << EOF
Name: libgbm-stub
Version: 0.3
Description: GBM stub for linking at build time
Requires: $REQUIRES

devel=\\
/wl/include/gbm.h \\
/wl/lib/libgbm.so \\
/wl/lib/pkgconfig/gbm.pc

exec=\\
/wl/lib/libstubgbm.so.1
EOF

exit
--- libgbm-stub-0.3.orig/Configure
+++ libgbm-stub-0.3/Configure
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^prefix/s|=.*|=$PREFIX|" gbm.pc
